content/blog/2020-01-13-a-comparison-of-javascript-typesystems.md

```diff
@@ -69,18 +82,23 @@ Flow offers the any / Object / Function types which essentially opt out of type-
 [Flow also has a repository for packages that don’t have their own typings](https://github.com/flow-typed/flow-typed). React comes shipped with flow types as it’s a Facebook project.
 
 ### ReasonML
-ReasonML is fully sound and complete. Type safety is build into the language. An additional benefit of having type safety build into the language is that it allows us to use types to communicate business logic. For an example of this we can look at the documentation of Rust (similar to ReasonML) [which shows a WebEvent](https://doc.rust-lang.org/stable/rust-by-example/custom_types/enum.html) and can at compile time tell us whether all possible states of this event are handled. ReasonML offers us these tools as well by not having a `null` type but providing the `Option` enum instead which can help enforce proper error handling.
+
+ReasonML is [fully sound](https://reasonml.github.io/docs/en/what-and-why#why-reason) and provides 100% coverage. Type safety is build into the language. An additional benefit of having type safety build into the language is that it allows us to use types to communicate business logic. For an example of this we can look at the documentation of Rust (similar to ReasonML) [which shows a WebEvent](https://doc.rust-lang.org/stable/rust-by-example/custom_types/enum.html) and can at compile time tell us whether all possible states of this event are handled. ReasonML offers us these tools as well by not having a `null` type but providing the `Option` enum instead which can help enforce proper error handling.
```

Comment:
Another example: https://yawaramin.github.io/re-web/re-web/Header/CacheControl/index.html#type-t

This type exactly models a cache-control response header and prevents creating nonsensical values like `no-store; max-age=1000`